technique/sans/instrument.py

```diff
@@ -107,15 +107,18 @@ def _generic_scan(self, detector, spectra, wiring, tcbs):
         """
         gen.change(nperiods=1)
         gen.change_start()
-        if self.get_pv("DAE:DETECTOR_FILE") != wiring:
+        for tcb in tcbs:
+            gen.change_tcb(**tcb)
+        gen.change_finish()
+        gen.change_start()
```

Comment:
Is there any particular reason you are changing tcbs separate to wiring tables? the memory usage required in the dae depends on both tcb and wiring, if you change them separately there is a chance the request could be rejected because the tcb for example are incompatible with the current wiring tables, but all would be ok if they were changed together. It may be in sans they never get close to these limits though.